OVS: Support OVS `netdev` datapath

cathay4t opened the pull request:
The OVS `netdev` datapath will use TUN interface for OVS internal
interface, nmstate should merge this kernel TUN interface state with
user space OVS interface state, otherwise verification failure will
happen.

The `netdev` datapath is used for OVS DPDK.  Manually tested with OVS 
DPDK on Intel E810-C card.


Integration test case included.

cathay4t commented:
The rpmbuild failure of epel-8 and 9 is caused by outdated rust. It will be automatically resolved once RHEL 9.2 released.

liangwen12year reviewed (commented):
A nit, not sure if it is possible to provide a querying example.

cathay4t committed:
OVS: Support OVS `netdev` datapath

The OVS `netdev` datapath will use TUN interface for OVS internal
interface, nmstate should merge this kernel TUN interface state with
user space OVS interface state, otherwise verification failure will
happen.

The `netdev` datapath is used for OVS DPDK. Manually tested with OVS
DPDK on Intel E810-C card.

Integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
> A nit, not sure if it is possible to provide a querying example.

Nothing special, the tun interface is still show as `ovs-interface`.

cathay4t commented:
Anyway, this is the example YAML(only used for reproduce the problem, not a valid use case, should use DPDK instead):

```yml
---
interfaces:
- name: ovs0
  type: ovs-interface
  ipv4:
    enabled: true
    dhcp: false
    address:
      - ip: 192.168.9.9
        prefix-length: 24
  ipv6:
    enabled: false
    dhcp: false
    autoconf: false
- name: br0
  type: ovs-bridge
  state: up
  bridge:
    options:
      datapath: "netdev"
    port:
    - name: ovs0
    - name: ovs1
ovs-db:
  other_config:
    dpdk-init: false
```
